vividus-bdd-engine/src/main/java/org/vividus/bdd/transformer/FilteringTableTransformer.java

```diff
@@ -65,7 +66,7 @@ public String transform(String tableAsString, ExamplesTableProperties properties
                 ? headerValues.size() : Math.min(headerValues.size(), Integer.parseInt(byMaxColumns));
         List<String> filteredColumnsNames = byColumnNames == null ? headerValues.subList(0, columnsLimit)
                 : Arrays.stream(StringUtils.split(byColumnNames, ';')).map(String::trim).collect(Collectors.toList());
-        Set<String> filteredHeaders = new HashSet<>(filteredColumnsNames);
+        Set<String> filteredHeaders = new TreeSet<>(filteredColumnsNames);
```

Comment:
I think, we should keep the same column order as it was before filtering, what about using `LinkedHashSet` ?